src/apps/vpn/platforms/linux/daemon/dbusservice.cpp

```diff
@@ -263,3 +284,58 @@ bool DBusService::firewallClear() {
   m_excludedApps.clear();
   return true;
 }
+
+/* Checks to see if the caller has sufficient authorization */
+bool DBusService::checkCallerAuthz() {
+  if (!calledFromDBus()) {
+    // That's unexpected, I don't think we should trust this.
+    return false;
+  }
+
+  // Get the PID of the D-Bus message sender.
+  QDBusConnectionInterface* iface = QDBusConnection::systemBus().interface();
+  QDBusReply<uint> reply = iface->servicePid(message().service());
+  uint senderpid = reply.value();
+  if (!reply.isValid() || (senderpid == 0)) {
+    // Could not lookup the sender's PID. Rejected!
+    logger.warning() << "Failed to resolve sender PID";
+    return false;
+  }
+
+  // Allow the call if it was made by the same binary.
+  if (!m_daemonExePath.isEmpty() &&
```

Comment:
This can be an issue for local development. Often i use the daemon from the deb package with a compiled client app. 

Would that work?

Reply:
Unfortunately, this usecase does not work without a step that requires root priveleges. Either the client must be installed first (eg: `sudo cmake --install <build-dir>`) or we must grant it the `CAP_NET_ADMIN` permission (eg: `sudo setcap cap_net_admin+ep <build-dir>/src/mozillavpn`)